### PR TITLE
[sparkle] - enh: `Tooltip`

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR aims at shipping the new `Tooltip` component from this PR https://github.com/dust-tt/dust/pull/7475

**References:**
 - https://github.com/dust-tt/dust/pull/7475

## Risk

Low

## Deploy Plan

Publish `sparkle`

### CODE REVIEW

File: sparkle/src/components/BarHeader.tsx [hunk 1]
```diff
@@ -97,7 +97,7 @@ BarHeader.ButtonBar = function (props: BarHeaderButtonBarProps) {
           icon={XMarkIcon}
           variant="tertiary"
           label="Close"
-          tooltipPosition="below"
+          tooltipPosition="bottom"
```
Comment (flvndvd) [inline sparkle/src/components/BarHeader.tsx:100]: I thought we were removing the position 🤔 ?

File: sparkle/src/components/Tooltip.tsx [hunk 2]
```diff
@@ -1,86 +1,38 @@
...
+      className || ""
+    )}
+    {...props}
+  />
+));
+TooltipContent.displayName = TooltipPrimitive.Content.displayName;
+
+export {
+  TooltipContent,
+  TooltipPortal,
```
Comment (flvndvd) [inline sparkle/src/components/Tooltip.tsx:34]: ✂️

File: sparkle/src/components/Breadcrumbs.tsx [hunk 3]
```diff
@@ -105,10 +105,7 @@ export function Breadcrumbs({ items }: BreadcrumbProps) {
 
 function truncateWithTooltip(text: string, length: number) {
   return text.length > length ? (
-    <Tooltip
-      label={text}
-      position="below"
-    >{`${text.substring(0, length - 1)}…`}</Tooltip>
+    <Tooltip trigger={`${text.substring(0, length - 1)}…`} label={text} />
```
Comment (flvndvd) [inline sparkle/src/components/Breadcrumbs.tsx:108]: `nit` We should use css for the ellipsis: `text-ellipsis`.

File: sparkle/src/components/ElementDialog.tsx [hunk 4]
```diff
@@ -33,6 +34,8 @@ export function ElementDialog<T>({
       onCancel={() => onCancel(transitionOnClose)}
       onValidate={() => onValidate(transitionOnClose)}
       {...props}
-    />
+    >
+      {children}
```
Comment (flvndvd) [inline sparkle/src/components/ElementDialog.tsx:38]: Why this change?

File: sparkle/src/components/IconButton.tsx [hunk 5]
```diff
@@ -1,16 +1,17 @@
...
-import { Tooltip, TooltipProps } from "./Tooltip";
+import { Tooltip } from "./Tooltip";
 
 type IconButtonProps = {
   variant?: "primary" | "warning" | "secondary" | "tertiary" | "white";
   onClick?: MouseEventHandler<HTMLButtonElement>;
   size?: "xs" | "sm" | "md";
   tooltip?: string;
-  tooltipPosition?: TooltipProps["position"];
+  tooltipPosition?: (typeof SIDE_OPTIONS)[number];
```
Comment (flvndvd) [inline sparkle/src/components/IconButton.tsx:14]: `nit` but I'd avoid spreading the underlying implementation of the `Tooltip` and would use `React.ComponentProps<Tooltip>["side"]`

**Review by flvndvd — APPROVED** [review]
This looks great 🔥🔥🔥

File: sparkle/src/components/ElementDialog.tsx (hunk 4 above)
Comment (JulesBelveze) [reply sparkle/src/components/ElementDialog.tsx:38]: My linter was actually warning `Element Dialog doesn't have required attribute children`

File: sparkle/src/components/Breadcrumbs.tsx (hunk 3 above)
Comment (JulesBelveze) [reply sparkle/src/components/Breadcrumbs.tsx:108]: We won't have control over the truncation length though?
